cpp/include/Ice/InputStream.h

```diff
@@ -1056,6 +1050,9 @@ class ICE_API InputStream : public IceInternal::Buffer
 
     void initialize(const EncodingVersion&);
 
+    // Reads a reference from the stream; the return value can be null.
+    IceInternal::ReferencePtr readReference();
```

Comment:
Because we're reading this "encoded proxy" as a Reference and returning a Reference. 

It's a private function so the exact name is not really critical.